packages/mongo-nav/src/MongoNav.tsx

```diff
@@ -111,27 +136,55 @@ export default function MongoNav({
   urls: urlsProp,
   constructOrganizationURL: constructOrganizationURLProp,
   constructProjectURL: constructProjectURLProp,
+  onError = () => {},
+  onSuccess = () => {},
 }: MongoNavInterface) {
-  let data;
+  const [data, setData] = React.useState<DataInterface | undefined>(undefined);
 
-  if (mode === Mode.Dev) {
-    data = devModeData;
+  const hosts = defaultsDeep(hostsProp, defaultHosts);
+  const endpointURI = `${hosts.cloud}/user/shared`;
+
+  function getProductionData() {
+    return fetch(endpointURI, {
+      credentials: 'include',
+      method: 'GET',
+    });
   }
 
-  if (!data) {
-    // Eventually this logic will be more robust, but for an alpha version will return null without data
-    return null;
+  function getFixtureData() {
+    return new Promise(resolve => {
+      onSuccess?.(fixtureData);
+
+      resolve(fixtureData);
+    });
   }
 
-  const {
-    account,
-    currentOrganization,
-    currentProject,
-    organizations,
-    projects,
-  } = data;
+  async function handleResponse(response: Response) {
+    if (!response.ok) {
+      const status = response.status as 401; //typecasting for now until we have more types to handle
+      onError?.(ErrorCodeMap[status]);
+      console.error(ErrorCodeMap[status]);
+    } else {
+      const data = await response.json();
+      setData(data);
+      onSuccess?.(data);
+    }
+  }
 
-  const hosts = defaultsDeep(hostsProp, defaultHosts);
+  useEffect(() => {
+    if (mode === Mode.Dev) {
+      getFixtureData().then(data => setData(data as DataInterface));
+    } else {
+      getProductionData()
+        .then(handleResponse)
+        .catch(console.error);
```

Comment:
I can believe a future where we gracefully fail and just show a top nav with the mongo nav, rather than nothing.

Reply:
Right now displaying the `<Loading />` component, but waiting on this https://jira.mongodb.org/browse/EXPO-254 before implementing any further logic